src/ox_thrift_server.erl

```diff
@@ -245,10 +245,10 @@ encode (#ts_config{service_module=ServiceModule, stats_module=StatsModule}, Prot
 
 
 -spec detect_protocol(Buffer::binary(), Default::atom()) -> atom().
-detect_protocol (<<16#82, _/binary>>, _Default)       -> ox_thrift_protocol_compact;
-detect_protocol (<<16#80, 16#1, _/binary>>, _Default) -> ox_thrift_protocol_binary; %% binary version 1
-detect_protocol (<<0, 0, _/binary>>, _Default)        -> ox_thrift_protocol_binary; %% binary version 0
-detect_protocol (_, Default)                          -> Default.
+detect_protocol (_, Protocol) when Protocol =/= undefined -> Protocol;
+detect_protocol (<<16#82, _/binary>>, undefined)          -> ox_thrift_protocol_compact;
+detect_protocol (<<16#80, 16#1, _/binary>>, undefined)    -> ox_thrift_protocol_binary; %% binary version 1
+detect_protocol (<<0, 0, _/binary>>, undefined)           -> ox_thrift_protocol_binary. %% binary version 0
```

Comment:
I'm not sure, if that will do what you want? After changes:
* when you specify `Protocol`, then that `Protocol` will be used **regardless** of `Buffer` decoding;
* you have to **unset** `Protocol` for it to be auto-deteced between the two.

Just want to make sure it's in line with others comments, PR desc and intentions.

Reply:
yep, before the PR, Default was never used, so setting protocol_module is kind of useless
I just wanted a way to force use another module like in https://github.com/openx/delivery-market-opportunity-server/pull/724

Reply:
Yep, I was also missing example usage :) now I see